### PR TITLE
💄 👻 add skeleton for soul-event text using `text_display`s

### PR DESCRIPTION
**Pull request by WesLSmith**
call for help gif
 
 
This pull request replaces chat window based call for help message with a basic block display based call for help messages.

Termination of this is currently handled by a temporary kill command in as_root.mcfunction

Followup tickets will be submitted for the following
1. Animate the unfurling of text as it is done in Undertale
2. Add undertale font pack to repository and update displays to point to that font pack
3. Generalize call for help display to be a function available for all soul events and terminate on reaching healing phase of event
4. Move call for help display to below TV (when it has been created) and adjust transformation matrix as appropriate.

### CODE REVIEW

File: datapacks/omega-flowey/data/entity/function/soul/soul_0/act_button/touch.mcfunction [hunk 1]
```diff
@@ -5,7 +5,9 @@ scoreboard players set #soul_0.touched soul.flag 1
...
 
 # TODO(82): this could be a lot cleaner visually
-say * You called for help...
+
+summon block_display -2.135 49.388 -33.859 {Tags:[call-for-help-display], Passengers:[{id:"minecraft:text_display", Tags:[call-for-help-display],  text:"{\"text\":\"* You Called For Help. . .\",\"color\":\"#FFFFFF\",\"bold\":false,\"italic\":false,\"underlined\":false,\"strikethrough\":false,\"font\":\"minecraft:uniform\"}",text_opacity:255,background:-1,alignment:"center",line_width:210,default_background:false, Rotation:[-180f,0f], transformation:[30.0150f,0.0000f,0.0000f,0.0000f,0.0000f,30.0700f,0.0000f,0.4900f,0.0000f,0.0000f,30.0100f,-0.0100f,0.0000f,0.0000f,0.0000f,1.0000f]},{id:"minecraft:text_display", Tags:[call-for-help-display], text:"{\"text\":\"* You Called For Help. . .\",\"color\":\"#FFFFFF\",\"bold\":false,\"italic\":false,\"underlined\":false,\"strikethrough\":false,\"font\":\"minecraft:uniform\"}",text_opacity:255,background:-16777216,alignment:"center",line_width:210,default_background:false, Rotation:[-180f,0f], transformation:[30.0000f,0.0000f,0.0000f,0.0000f,0.0000f,30.0000f,0.0000f,0.5000f,0.0000f,0.0000f,30.0000f,0.0000f,0.0000f,0.0000f,0.0000f,1.0000f]}]}
```
Comment (TheAfroOfDoom) [inline datapacks/omega-flowey/data/entity/function/soul/soul_0/act_button/touch.mcfunction:9]: we can use the trailing backslashes to wrap this summon command so it's much more readable.

```mcfunction
summon block_display -2.135 49.388 -33.859 {\
  Tags:[call-for-help-display], \
  Passengers:[\
  ...
```

most commands in the repo aren't long enough to be worth it but this one definitely is

Comment (TheAfroOfDoom) [reply datapacks/omega-flowey/data/entity/function/soul/soul_0/act_button/touch.mcfunction:9]: it shows up poorly in GitHub but with spyglass in vscode it will look fine

Comment (TheAfroOfDoom) [inline datapacks/omega-flowey/data/entity/function/soul/soul_0/act_button/touch.mcfunction:9]: these xyz coords seem kind of arbitrary. I guess they don't matter a ton since they will be summoned at the TV in the other ticket?

even still id prefer they were ~roundish numbers now:

```mcfunction
summon block_display -2.5 49.5 -34.0
```

**Review by TheAfroOfDoom — CHANGES_REQUESTED** [review]
more 2 come tomorrow

File: datapacks/omega-flowey/data/entity/function/soul/soul_0/act_button/touch.mcfunction (hunk 1 above)
Comment (TheAfroOfDoom) [reply datapacks/omega-flowey/data/entity/function/soul/soul_0/act_button/touch.mcfunction:9]: once https://github.com/MinecraftCommands/syntax-mcfunction/issues/17 is resolved it will show up nicer in Github

Comment (WesLSmith) [reply datapacks/omega-flowey/data/entity/function/soul/soul_0/act_button/touch.mcfunction]: Updated this to be a bit nicer, also took out some undeeded arguements

Comment (TheAfroOfDoom) [reply datapacks/omega-flowey/data/entity/function/soul/soul_0/act_button/touch.mcfunction]: we could also wrap the transformation matrix so that it looks closer to the 4x4 matrix that it is instead of being flat

Comment (WesLSmith) [reply datapacks/omega-flowey/data/entity/function/soul/soul_0/act_button/touch.mcfunction]: done

File: datapacks/omega-flowey/data/entity/function/soul/soul_0/act_button/touch.mcfunction [hunk 2]
```diff
@@ -5,7 +5,41 @@ scoreboard players set #soul_0.touched soul.flag 1
 function animated_java:act_button/variants/selected/apply
 
 # TODO(82): this could be a lot cleaner visually
-say * You called for help...
+
+summon block_display -2.5 49.5 -34.0 \ 
+    {Tags:[call-for-help-display], \
```
Comment (TheAfroOfDoom) [inline datapacks/omega-flowey/data/entity/function/soul/soul_0/act_button/touch.mcfunction:10]: any entity summoned by the map as a whole (anything in the code) have tags that categorize the entity at multiple levels. for example, see soul 0's `executor`'s tags

https://github.com/TheAfroOfDoom/omega-flowey-minecraft-remastered/blob/596c4842bd3007412d8ac6749cd727217460448f/datapacks/omega-flowey/data/entity/function/soul/soul_0/start.mcfunction#L2

<details>
<summary>(terminology: executor)</summary>

an `executor` is the thing that runs the primary/high level logic for an event/attack. e.g. so when you start a soul event, the first thing that happens is we summon an `executor`

</details>

these tags all serve a purpose:

## "omega-flowey-remastered"

this is the global tag that any entity summoned by the map will have. this is required so that we can run logic on that entity every tick.

https://github.com/TheAfroOfDoom/omega-flowey-minecraft-remastered/blob/596c4842bd3007412d8ac6749cd727217460448f/datapacks/omega-flowey/data/omega-flowey/function/tick.mcfunction#L3-L5

this line calls each entity with the tag (`@e[tag=omega-flowey-remastered]`) and runs a heavily nested filtering function on it until it gets to its entity-specific code.

https://github.com/TheAfroOfDoom/omega-flowey-minecraft-remastered/blob/596c4842bd3007412d8ac6749cd727217460448f/datapacks/omega-flowey/data/entity/function/tick.mcfunction#L1-L3

https://github.com/TheAfroOfDoom/omega-flowey-minecraft-remastered/blob/596c4842bd3007412d8ac6749cd727217460448f/datapacks/omega-flowey/data/entity/function/soul/tick.mcfunction#L1

https://github.com/TheAfroOfDoom/omega-flowey-minecraft-remastered/blob/596c4842bd3007412d8ac6749cd727217460448f/datapacks/omega-flowey/data/entity/function/soul/soul_0/tick.mcfunction#L1-L5

this is basically the required and efficient way to run logic on an entity every tick. (there are 20 (game)ticks per second in Minecraft, 0.05 s/tick)

the `soul` and `soul-executor` tags serve a similar purpose (along with some other things) that you could search for if you're curious with `tag=soul`/`tag=soul-executor`.

## "soul-executor-new"

this is exclusively to initialize the entity you just summoned. you'll see this pattern everywhere:

https://github.com/TheAfroOfDoom/omega-flowey-minecraft-remastered/blob/596c4842bd3007412d8ac6749cd727217460448f/datapacks/omega-flowey/data/entity/function/soul/soul_0/start.mcfunction#L1-L5

https://github.com/TheAfroOfDoom/omega-flowey-minecraft-remastered/blob/596c4842bd3007412d8ac6749cd727217460448f/datapacks/omega-flowey/data/entity/function/soul/soul_0/executor/initialize.mcfunction#L14-L20

if your entity doesn't need any initialization, you technically dont need a `"...-new"` tag. but i usually add one and a blank initialization function (that removes the new tag) anyway just so its easy to add initialization to it later if i end up needing to. it's not really a performance concern

## "soul_0"

this is also used for tick filtering as mentioned above in `omega-flowey-remastered`, but more importantly for the display entity we can use it for a cleaner termination function:

https://github.com/TheAfroOfDoom/omega-flowey-minecraft-remastered/blob/596c4842bd3007412d8ac6749cd727217460448f/datapacks/omega-flowey/data/entity/function/soul/soul_0/executor/terminate.mcfunction#L2-L3

https://github.com/TheAfroOfDoom/omega-flowey-minecraft-remastered/blob/160cceab44ccd8ed46a2a6a32cfd084015eb56b0/datapacks/omega-flowey/data/entity/function/soul/soul_0/executor/terminate/as_root.mcfunction#L1-L3

every `bullet`, `bandaid`, and the `act-button` have the `soul_0` tag, so the `executor` only has to use `@e` once and have each applicable entity call the `as_root` function, which does the self-termination as you see above based on their identifying tag

we can just add another line to that list in `as_root` to more cleanly terminate the display

## TL;DR

for brevity ill just say to give this new entity at least these tags:

```
"omega-flowey-remastered","soul","soul_0","call-for-help-display"
```

Comment (TheAfroOfDoom) [reply datapacks/omega-flowey/data/entity/function/soul/soul_0/act_button/touch.mcfunction:10]: on a similar note, we should add a new entity folder under `soul_0` for the display. the main ones it looks like it'll need are a `summon` function and `terminate` function.

```
soul
  soul_0
    act_button
    bandaid
    ...
    help_display
      summon.mcfunction
      terminate.mcfunction
    locator
    ...
```

these are nice to have so we have a single source of truth for each of those behaviors.

File: datapacks/omega-flowey/data/entity/function/soul/soul_0/act_button/touch.mcfunction [hunk 3]
```diff
@@ -5,7 +5,41 @@ scoreboard players set #soul_0.touched soul.flag 1
 function animated_java:act_button/variants/selected/apply
 
 # TODO(82): this could be a lot cleaner visually
-say * You called for help...
+
+summon block_display -2.5 49.5 -34.0 \ 
+    {Tags:[call-for-help-display], \ 
+    Passengers:[ \
```
Comment (TheAfroOfDoom) [inline datapacks/omega-flowey/data/entity/function/soul/soul_0/act_button/touch.mcfunction:11]: it looks like there are 2 passengers here that are pretty much the exact same. i suspect this is unintentional and the display will look identical if you remove one of these passengers

File: datapacks/omega-flowey/data/entity/function/soul/soul_0/act_button/touch.mcfunction (hunk 2 above)
Comment (TheAfroOfDoom) [inline datapacks/omega-flowey/data/entity/function/soul/soul_0/act_button/touch.mcfunction:10]: i have some nits about the general style of wrapping decisions BUT i think its unreasonable to list them since we should just have an auto formatter eventually once Spyglass implements it

**Review by TheAfroOfDoom — CHANGES_REQUESTED** [review]

File: datapacks/omega-flowey/data/entity/function/soul/soul_0/act_button/touch.mcfunction (hunk 3 above)
Comment (WesLSmith) [reply datapacks/omega-flowey/data/entity/function/soul/soul_0/act_button/touch.mcfunction]: This is intentional - this is how we create the white border to the black backgrounded text box. A slightly large white tex/white background box with the same length text is placed behind the black background/white text box to give the apperance of there being a border

Comment (WesLSmith) [reply datapacks/omega-flowey/data/entity/function/soul/soul_0/act_button/touch.mcfunction]: I think we could go even further and project this 2d surface onto a display_block if we wanted to give it some depth

**Review by TheAfroOfDoom — APPROVED** [review]
thx
